src/chart/line/lineAnimationDiff.js

```diff
@@ -166,6 +166,13 @@ export default function (
 
     // Diff result may be crossed if all items are changed
     // Sort by data index
+    for (var i = rawIndices.length - 1; i >= 0; i--) {
+        for (var j = 0; j < rawIndices.length; j++) {
+            if (rawIndices[i] === rawIndices[j]) {
+                rawIndices.splice(i, 1);
+            }
+        }
+    }
```

Comment:
O(n2) algorithm may have performance issue on large data. 

Also splice is usually not a performant method and can easily get logic messed up(because the array length is changed after splice). Create a new array(when necessary) is much simpler and has better performance. Check https://jsperf.com/splice-vs-filter 

Since there `rawIndices` has been sorted in the following code. We can loop sortedIndices, and only compare two neighbor rawIndex to see if they are the same.